### DISCUSSION TITLE
Horrible integration in Firefox

### DISCUSSION THREAD

**Discussion by usernamesaredumb2** [aclap-dev/video-downloadhelper#225] [Help and Q&A]
### Your question

DownloadHelper no longer integrates into Firefox properly.
It doesn't respect the Firefox setting to NOT create any history at all. Instead, a history of downloaded videos is now created and must be wiped manually.
Also, active and finished downloads no longer show up in the Firefox downloads page. Instead you have to guess colors of stupid dots in the DownloadHelper GUI to open a popup of currently running downloads. That's NOT a proper replacement for the excellent Firefox downloads page, which can stay open all the time in an extra tab and gives you a perfect overview of currently running and recently finished downloads. The original downloads page also provided an easy mechanism to resume cancelled downloads: Just restart the download, pause it, replace the partially downloaded file with the cancelled one, resume and Firefox picked up the download where it was cancelled. DownloadHelper's custom GUI doesn't even provide a pause button.

Usability got so much worse with the "oh we can do it all alone, we don't need the Firefox GUI at all" update.
Used to be such a great plugin.
Just WHY??

### Addon & CoApp version and details

8.2.1.1

**Comment by Mattmjs** [comment]
I'm another user of the software but I'll try to explain how it behaves with Firefox and without. When you come across a video
which is not chunked steaming ( it's a single mp4 file ) you can use the browser to handle the download.
I use the browser on these. But these are the only type of video that the browser can handle on it's own.
If you come across a chunked video stream where the video is stream as small pieces then the browser cannot handle these.
They can only be handled by the companion app, hence you have no browser downloading option.
When you see **HLS**  and **Dash** in the extension menu, that means they are chunked streams. Nothing has changed, it has been
like this since the extension required the companion app. What history are you talking about ?

**Comment by paulrouget** [comment]
I think @usernamesaredumb2 is talking about a naming bug we recently introduced.

@usernamesaredumb2 in the settings, check if "Behavior > Download processor" is correctly set to Browser. This will make sure it's using Firefox UI for download (and of course, like @Mattmjs said, that only works for regular files, not HLS/Dash).

If this doesn't solve your issue, please share a link to the page you're trying to download from, and I'll see what's wrong.

**Comment by usernamesaredumb2** [comment]
I'm not talking about HLS streams, the companion app or any naming bugs, just ordinary video downloads like the following:
https://www.justwatch.com/de/Serie/Yellowstone-2018
When downloading the file in DownloadHelper, it doesn't show up in "Extras > Downloads" (Firefox 115.7.0esr (64-Bit)).
However, the downloaded files appear in Firefox' history of visited pages, despite having configured Firefox to never create a history at all (it doesn't store any pages in the browser history, just the downloaded files from DownloadHelper).
"Behavior > Download processor" is set to Browser, I don't have the companion app.

**Comment by paulrouget** [comment]
> Usability got so much worse with the "oh we can do it all alone, we don't need the Firefox GUI at all" update. Used to be such a great plugin.
> Just WHY??

It's a bug. It happens :)

Let's try to figure this out.

First, can I ask you to install the [Beta version](https://downloadhelper.net/beta), to make sure you're as up to date as possible (you can revert to regular version once we fixed your problem).

> it doesn't show up in "Extras > Downloads"

You mean, in that screen?

<img width="610" alt="Screenshot 2024-03-01 at 14 26 01" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/a8f76dbc-3ff7-474d-bc94-e3da1597b000">

> However, the downloaded files appear in Firefox' history of visited pages, despite having configured Firefox to never create a history at all (it doesn't store any pages in the browser history, just the downloaded files from DownloadHelper).

I think you configured Firefox for incognito mode by default, is that correct?

**Comment by paulrouget** [comment]
@usernamesaredumb2 I have pushed Beta 8.2.1.7a1: https://downloadhelper.net/data/vdh-mozilla-beta-latest.xpi

Can you install it and tell me if it addresses some of your concerns?
